[bitnami/kafka] Provisioning of topic doesn't support bundle CA

### Name and Version

bitnami/kafka 21.4.4

### What architecture are you using?

amd64

### What steps will reproduce the bug?

This issue can be reproduced with bundle CA cert on provisioning client side.

1. Copy kust-secret.yaml to bitnami/kafka/templates (in overrides.zip)
2. Copy bitnami/common and bitnami/zookeeper to bitnami/kafka/charts
3. Deploy with command `helm install -n ns-kafka --create-namespace kafka -f values-override.yaml .` (values-override.yaml in overrides.zip)
4. Check provisioning container log `kubectl -n ns-kafka logs -f kafka-provisioning-7vt9x` (the pod name should be different than yours)

[overrides.zip](https://github.com/bitnami/charts/files/11212795/overrides.zip)


### Are you using any custom parameters or values?

Yes. In overrides.zip uploaded in the above section.

### What is the expected behavior?

Provision of topic can successfully run.

### What do you see instead?

Get error.

> Defaulted container "kafka-provisioning" out of: kafka-provisioning, wait-for-available-kafka (init)
> Configuring environment
> Running pre-provisioning script if any given
> Starting provisioning
> [2023-04-12 15:02:25,522] ERROR [AdminClient clientId=adminclient-1] Connection to node -1 (kafka/10.43.38.70:9092) failed authentication due to: SSL handshake failed (org.apache.kafka.clients.NetworkClient)
> [2023-04-12 15:02:25,524] WARN [AdminClient clientId=adminclient-1] Metadata update failed due to authentication error (org.apache.kafka.clients.admin.internals.AdminMetadataManager)
> org.apache.kafka.common.errors.SslAuthenticationException: SSL handshake failed
> Caused by: javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
>         at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
>         at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:353)
>         at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:296)
>         at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:291)
>         at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1357)
>         at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1232)
>         at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1175)
>         at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
>         at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
>         at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1076)
>         at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1063)
>         at java.base/java.security.AccessController.doPrivileged(Native Method)
>         at java.base/sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:1010)
>         at org.apache.kafka.common.network.SslTransportLayer.runDelegatedTasks(SslTransportLayer.java:435)
>         at org.apache.kafka.common.network.SslTransportLayer.handshakeUnwrap(SslTransportLayer.java:523)
>         at org.apache.kafka.common.network.SslTransportLayer.doHandshake(SslTransportLayer.java:373)
>         at org.apache.kafka.common.network.SslTransportLayer.handshake(SslTransportLayer.java:293)
>         at org.apache.kafka.common.network.KafkaChannel.prepare(KafkaChannel.java:178)
>         at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:543)
>         at org.apache.kafka.common.network.Selector.poll(Selector.java:481)
>         at org.apache.kafka.clients.NetworkClient.poll(NetworkClient.java:560)
>         at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.processRequests(KafkaAdminClient.java:1413)
>         at org.apache.kafka.clients.admin.KafkaAdminClient$AdminClientRunnable.run(KafkaAdminClient.java:1344)
>         at java.base/java.lang.Thread.run(Thread.java:829)
> Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
>         at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
>         at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
>         at java.base/sun.security.validator.Validator.validate(Validator.java:264)
>         at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
>         at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:276)
>         at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:141)
>         at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1335)
>         ... 19 more
> Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
>         at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:141)
>         at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:126)
>         at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
>         at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
>         ... 25 more
> Error while executing topic command : SSL handshake failed


### Additional information

In the secret, especially kafka-client-tls-secret, in the uploaded kust-secret.yaml, when

- Remove the second CA cert (root CA)
- Exchange the two certs

Provisioning can work.

The fix of it is: Instead of inlining the certs in the config file, reference the cert file locations in the config file.

--- a/bitnami/kafka/Chart.yaml
+++ b/bitnami/kafka/Chart.yaml
@@ -29,4 +29,4 @@ name: kafka
 sources:
   - https://github.com/bitnami/containers/tree/main/bitnami/kafka
   - https://kafka.apache.org/
-version: 21.4.5
+version: 21.4.6

--- a/bitnami/kafka/templates/kafka-provisioning.yaml
+++ b/bitnami/kafka/templates/kafka-provisioning.yaml
@@ -108,12 +108,17 @@ spec:
                 kafka_common_conf_set "$CLIENT_CONF" ssl.truststore.type {{ upper .Values.provisioning.auth.tls.type | quote }}
                 ! is_empty_value "$KAFKA_CLIENT_KEY_PASSWORD" && kafka_common_conf_set "$CLIENT_CONF" ssl.key.password "$KAFKA_CLIENT_KEY_PASSWORD"
                 {{- if eq (upper .Values.provisioning.auth.tls.type) "PEM" }}
+                {{- if .Values.provisioning.auth.tls.caCert }}
                 file_to_multiline_property() {
                     awk 'NR > 1{print line" \\"}{line=$0;}END{print $0" "}' <"${1:?missing file}"
                 }
                 kafka_common_conf_set "$CLIENT_CONF" ssl.keystore.key "$(file_to_multiline_property "/certs/{{ .Values.provisioning.auth.tls.key }}")"
                 kafka_common_conf_set "$CLIENT_CONF" ssl.keystore.certificate.chain "$(file_to_multiline_property "/certs/{{ .Values.provisioning.auth.tls.cert }}")"
                 kafka_common_conf_set "$CLIENT_CONF" ssl.truststore.certificates "$(file_to_multiline_property "/certs/{{ .Values.provisioning.auth.tls.caCert }}")"
+                {{- else }}
+                kafka_common_conf_set "$CLIENT_CONF" ssl.keystore.location "/certs/{{ .Values.provisioning.auth.tls.keystore }}"
+                kafka_common_conf_set "$CLIENT_CONF" ssl.truststore.location "/certs/{{ .Values.provisioning.auth.tls.truststore }}"
+                {{- end }}
                 {{- else if eq (upper .Values.provisioning.auth.tls.type) "JKS" }}
                 kafka_common_conf_set "$CLIENT_CONF" ssl.keystore.location "/certs/{{ .Values.provisioning.auth.tls.keystore }}"
                 kafka_common_conf_set "$CLIENT_CONF" ssl.truststore.location "/certs/{{ .Values.provisioning.auth.tls.truststore }}"

--- a/bitnami/kafka/values.yaml
+++ b/bitnami/kafka/values.yaml
@@ -1568,7 +1568,10 @@ provisioning:
       type: jks
       ## @param provisioning.auth.tls.certificatesSecret Existing secret containing the TLS certificates for the Kafka provisioning Job.
       ## When using 'jks' format for certificates, the secret should contain a truststore and a keystore.
-      ## When using 'pem' format for certificates, the secret should contain a public CA certificate, a public certificate and one private key.
+      ## When using 'pem' format for certificates, the secret should contain one of the following:
+      ## 1. A public CA certificate, a public certificate and one private key.
+      ## 2. A truststore and a keystore in PEM format
+      ## If caCert is set, option 1 will be taken, otherwise option 2.
       ##
       certificatesSecret: ""
       ## @param provisioning.auth.tls.cert The secret key from the certificatesSecret if 'cert' key different from the default (tls.crt)

